tests/unit/applications/test_wizard.py

```diff
@@ -35,6 +39,29 @@ def test_is_field_optional(complete_application):
     assert not wizard.is_field_optional("unknown_field", dummy_form_spec)
 
 
+def test_is_valid_false(incomplete_application):
+    wizard = Wizard(incomplete_application, [])
```

Comment:
Am a bit uneasy about the empty list here.  There's an unstated assumption in some of the rest of the code that there's a form_spec for each of an application's page model instances, and I don't know what would happen when this assumption does not hold.

I'd prefer it if the empty list was replaced by `form_specs`, as elsewhere in the tests.